src/hooks.d.ts

```diff
@@ -62,6 +68,8 @@ export interface ModelHooks<M extends Model = Model, TAttributes = any> {
   afterBulkSync(options: SyncOptions): HookReturn;
   beforeQuery(options: QueryOptions, query: AbstractQuery): HookReturn;
   afterQuery(options: QueryOptions, query: AbstractQuery): HookReturn;
+  beforeAssociate(data: BeforeAssociateEventData, options: AssociationOptions<any>): HookReturn;
```

Comment:
I'd prefer to go with the safer "unknown"

```suggestion
  beforeAssociate(data: BeforeAssociateEventData, options: AssociationOptions<unknown>): HookReturn;
```

Reply:
Makes sense, I figured out `unknown` is better for this case than `any`